src/abi_impl.rs

```diff
@@ -78,6 +78,37 @@ fn call_module(env: &Env, address: &Address, function: &str, param: &str) -> ABI
     }
 }
 
+/// Transfer an amount from the address on the current call stack to a target address.
+pub(crate) fn transfer_coins(env: &Env, to_address: i32, raw_amount: i64) -> ABIResult<()> {
+    sub_remaining_point(env, settings::metering_transfer())?;
+    let memory = get_memory!(env);
+    let to_address = &get_string(memory, to_address)?;
+    match env.interface.transfer_coins(to_address, raw_amount as u64) {
```

Comment:
conversion from i64 to u64 will panic if the amount is negative.
Let's check beforehand and bail! if the amount is negative